config/fcos/v1_5_exp/validate_test.go

```diff
@@ -231,3 +231,102 @@ func TestValidateGrubUser(t *testing.T) {
 		})
 	}
 }
+
+func TestValidateConfig(t *testing.T) {
+	tests := []struct {
+		config  Config
+		err     error
+		errPath path.ContextPath
+	}{
+		// valid config
+		{
+			config: Config{
+				Config: base.Config{
+					Storage: base.Storage{
+						Disks: []base.Disk{
+							{
+								Device:    "/dev/vda",
+								WipeTable: util.BoolToPtr(false),
+								Partitions: []base.Partition{
+									{
+										Label:  util.StrToPtr("foo"),
+										Number: 1,
+									},
+									{
+										Label:  util.StrToPtr("bar"),
+										Number: 2,
+									},
+								},
+							},
+						},
+					},
+				},
+			},
+			err:     nil,
+			errPath: path.New("yaml"),
+		},
+		// specify root disk
+		{
+			config: Config{
+				Config: base.Config{
+					Storage: base.Storage{
+						Disks: []base.Disk{
+							{
+								Device:    "/dev/disk/by-id/coreos-boot-disk",
+								WipeTable: util.BoolToPtr(false),
+								Partitions: []base.Partition{
+									{
+										Label:  util.StrToPtr("foo"),
+										Number: 0,
+									},
+									{
+										Label:  util.StrToPtr("bar"),
+										Number: 2,
+									},
+								},
+							},
+						},
+					},
+				},
+			},
+			err:     nil,
+			errPath: path.New("yaml"),
+		},
+		// invalid config
+		{
+			config: Config{
+				Config: base.Config{
+					Storage: base.Storage{
+						Disks: []base.Disk{
+							{
+								Device:    "/dev/vda",
+								WipeTable: util.BoolToPtr(false),
+								Partitions: []base.Partition{
+									{
+										Label:  util.StrToPtr("foo"),
+										Number: 0,
+									},
+									{
+										Label:  util.StrToPtr("bar"),
+										Number: 1,
+									},
+								},
+							},
+						},
+					},
+				},
+			},
+			err:     common.ErrInvalidPartitionNumber,
+			errPath: path.New("yaml"),
+		},
+	}
+
```

Comment:
this seems to cause the test to fail not sure where i'm going wrong here

Reply:
im getting this output when i run the test, its odd through becasue the other tests dont fail when an error message is in the err: field.

``` --- FAIL: TestValidateConfig (0.00s)
    --- FAIL: TestValidateConfig/validate_2 (0.00s)
        /home/adamobrien/coreos/coreos-butane/butane/config/fcos/v1_5_exp/validate_test.go:329: 
            	Error Trace:	/home/adamobrien/coreos/coreos-butane/butane/config/fcos/v1_5_exp/validate_test.go:329
            	Error:      	Not equal: 
            	            	expected: report.Report{Entries:[]report.Entry{report.Entry{Kind:0, Message:"unexpected partition number. Partition number should only be zero when using root disk or wipe_tables is true. Specify root disk as /dev/disk/by-id/coreos-boot-disk ", Context:path.ContextPath{Path:[]interface {}(nil), Tag:"yaml"}, Marker:tree.Marker{StartP:(*tree.Pos)(nil), EndP:(*tree.Pos)(nil)}}}}
            	            	actual  : report.Report{Entries:[]report.Entry{report.Entry{Kind:1, Message:"unexpected partition number. Partition number should only be zero when using root disk or wipe_tables is true. Specify root disk as /dev/disk/by-id/coreos-boot-disk ", Context:path.ContextPath{Path:[]interface {}{0}, Tag:"partition_number"}, Marker:tree.Marker{StartP:(*tree.Pos)(nil), EndP:(*tree.Pos)(nil)}}}}
            	            	
            	            	Diff:
            	            	--- Expected
            	            	+++ Actual
            	            	@@ -3,7 +3,9 @@
            	            	   (report.Entry) {
            	            	-   Kind: (report.Kind) 0,
            	            	+   Kind: (report.Kind) 1,
            	            	    Message: (string) (len=165) "unexpected partition number. Partition number should only be zero when using root disk or wipe_tables is true. Specify root disk as /dev/disk/by-id/coreos-boot-disk ",
            	            	    Context: (path.ContextPath) {
            	            	-    Path: ([]interface {}) <nil>,
            	            	-    Tag: (string) (len=4) "yaml"
            	            	+    Path: ([]interface {}) (len=1) {
            	            	+     (int) 0
            	            	+    },
            	            	+    Tag: (string) (len=16) "partition_number"
            	            	    },
            	Test:       	TestValidateConfig/validate_2
            	Messages:   	invalid report
FAIL
FAIL	github.com/coreos/butane/config/fcos/v1_5_exp	0.006s
FAIL
 ```

Reply:
I saw that it looked like you already fixed the test. But as you now know the Context.Path you gave expected 'yaml' but the code returned 'partition_number'.